### PR TITLE
Expand documentation for Table components

### PR DESCRIPTION
**Pull request by dgalarza**

### CODE REVIEW

File: packages/components/table/stories/table.stories.mdx [hunk 1]
```diff
@@ -75,6 +74,95 @@ export const Template = ({ cellProps = () => ({}), ...args }) => {
...
+    </tr>
+    <tr>
+      <td>cellProps (optional)</td>
+      <td>
+        Props to be passed to the table cell when rendered. They can be any props
+        that can be passed to `Box`.
+      </td>
+    </tr>
+    <tr>
+      <td>render</td>
```
Comment (acostanzo) [inline packages/components/table/stories/table.stories.mdx:124]:
```suggestion
      <td>render (optional)</td>
```

File: packages/components/table/stories/table.stories.mdx [hunk 2]
```diff
@@ -75,6 +74,95 @@ export const Template = ({ cellProps = () => ({}), ...args }) => {
...
+      <td>
+        A function used in order to render the row. If provided, the record will
+        be passed as the only argument to the render function. If no render function
+        is provided then td will attempt to read the dataKey from the object.
```
Comment (acostanzo) [inline packages/components/table/stories/table.stories.mdx:128]: This got me thinking... we got backed into a corner of positional arguments in the past when we wanted to support more features in the render function. Should we change the render functions argument to be an object? Updating its interface **from:**
```
render: (rowData) => rowData.foo
```
**To**
```
render: ({ rowData }) => rowData.foo
```

File: packages/components/table/stories/table.stories.mdx [hunk 3]
```diff
@@ -75,6 +74,95 @@ export const Template = ({ cellProps = () => ({}), ...args }) => {
...
+      <td>dataKey</td>
+      <td>
+        A unique identifier for the cell in a row. This can be propert of the
+        record being rendered or a nested property within the record being rendered.
+      </td>
+    </tr>
+    <tr>
+      <td>cellProps (optional)</td>
+      <td>
+        Props to be passed to the table cell when rendered. They can be any props
```
Comment (acostanzo) [inline packages/components/table/stories/table.stories.mdx:119]: Cell props takes a function similar to render so you have access to the rowData. If we make the recommended update below, I think we should do it for cellProps too for consistency

File: packages/components/table/stories/table.stories.mdx (hunk 2 above)
Comment (dgalarza) [reply packages/components/table/stories/table.stories.mdx:128]: I kinda prefer the first argument being the record and if we ever take additional args it being an options object that can then be destructed.

Comment (acostanzo) [reply packages/components/table/stories/table.stories.mdx:128]: Oh I like that, cool!

**Pull request merged**